### PR TITLE
[luci] Support Sub in NCHWToNHWCPass

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise to support Sub node in NCHWToNHWCPass.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #7308, from draft #7309

File: compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp [hunk 1]
```diff
@@ -823,6 +905,73 @@ class ConvertNCHWToNHWC final : public luci::CircleNodeMutableVisitor<bool>
...
+      auto nhwc_const = create_NHWC_from_NCHW(subtract);
+      if (nhwc_const == nullptr)
+        return false;
+
+      node->x(pre_trans);
+      node->y(nhwc_const);
+    }
+    else if (subtract == nullptr)
+    {
+      // Both inputs are not constant.
```
Comment (jinevening) [inline compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp:942]: Is this true? It seems that both `is_NCHW_with_const_x` and `is_NCHW_with_const_y` can return false even when one of x/y is const.

Comment (seanshpark) [reply compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp:942]: Hum... it was suspicious to me too... I'll revise this in another way...

File: compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp [hunk 2]
```diff
@@ -823,6 +905,77 @@ class ConvertNCHWToNHWC final : public luci::CircleNodeMutableVisitor<bool>
...
+      if (nhwc_const == nullptr)
+        return false;
+
+      node->x(pre_trans);
+      node->y(nhwc_const);
+    }
+    else
+    {
+      auto const_x = dynamic_cast<luci::CircleConst *>(node->x());
+      auto const_y = dynamic_cast<luci::CircleConst *>(node->y());
```
Comment (jinevening) [inline compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp:943]: This pattern appears three times in this PR. Actually, this information is used in all three cases.

I think the if statements can be restructured as follows.

```cpp
auto const_x = dynamic_cast<luci::CircleConst *>(node->x());
auto const_y = dynamic_cast<luci::CircleConst *>(node->y());

if (const_x == nullptr && const_y != nullptr)
{
  // Check NCHW for const_y
  // Insert pre-transpose
}
else if (const_x == nullptr && const_y != nullptr)
{
  // Check NCHW for const_x
  // Insert pre-transpose

}
else if (const_x == nullptr && const_y != nullptr)
{
  as-is
}
```

`post_NCHW_with_const` may be used for checking NCHW.

Comment (seanshpark) [reply compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp]: Thanks for the suggestion!

**Comment by seanshpark** [comment]
@jinevening , now the code looks better! :)

**Review by jinevening — APPROVED** [review]
LGTM Thank you

**Comment by seanshpark** [comment]
@nnfw-bot test nncc-debug

**Comment by seanshpark** [comment]
@nnfw-bot test nncc-release

**Pull request merged**